week-four/app/src/main/java/pro/marvinhosea/movielist/ui/movies/MainActivity.kt

```diff
@@ -0,0 +1,42 @@
+package pro.marvinhosea.movielist.ui.movies
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import pro.marvinhosea.movielist.R
+import pro.marvinhosea.movielist.ui.login.UserLoginActivity
+import pro.marvinhosea.movielist.repository.UserSharedPrefRepository
+
+class MainActivity : AppCompatActivity() {
+
+    private var userSharedRepository = UserSharedPrefRepository
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        setSupportActionBar(findViewById(R.id.toolbar))
+
+        userSharedRepository.init(this)
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        if (!userSharedRepository.isUserLoggedIn()) {
+            val intent = Intent(this, UserLoginActivity::class.java)
+            startActivity(intent)
+            finish()
+        }
+
+    }
+
+    override fun onResume() {
```

Comment:
I don't think you understand the lifecycle graph from what I see. the functions **always** happen one after another, in order.

So onStart will ALWAYS be called before onResume, just as well as onResume will ALWAYS go after onStart.

In this case, it doesn't make sense to have the same code in two lifecycle callbacks, as you're calling the same code twice.

Also, if you did plan to add it to both lifecycle callbacks, I'd have pulled out the code into a separate function, and reuse it, and not copy/paste it.